lib/elastic_record/relation.rb

```diff
@@ -75,13 +75,21 @@ def search_results
         @search_results ||= begin
           options = search_type_value ? {search_type: search_type_value} : {}
 
-          klass.elastic_index.search(as_elastic.update('_source' => false), options)
+          if klass.elastic_index.load_from_source
+            klass.elastic_index.search(as_elastic, options)
+          else
+            klass.elastic_index.search(as_elastic.update('_source' => false), options)
+          end
         end
       end
 
-      def load_hits(ids)
-        scope = select_values.any? ? klass.select(select_values) : klass
-        scope.find(ids)
+      def load_hits
+        if klass.elastic_index.load_from_source
+           search_hits.map { |hit| klass.new(hit['_source'].merge('id' => hit['_id'])) }
```

Comment:
`merge` is going to create a new hash for each record. You can probably use `update` instead here.

Reply:
or `merge!`. Not entirely sure which one's the alias based on the Ruby docs.

Reply:
I was trying to avoid modifying the `hits` objects in place.  I think it would be acceptable to do so if we make `search_hits` a private method.  It looks like it not called outside of `ElasticRecord::Relation`.